### DISCUSSION TITLE
Function Definition field operation data type string or object ?

### DISCUSSION THREAD

**Discussion by shahbaaz31here** [serverlessworkflow/specification#802] [Q&A]
Hi Team,
As per https://github.com/serverlessworkflow/specification/blob/main/specification.md#function-definition **FunctionDefinition.operation** can be String or Object type. 
In Serverless sdk 4.0.4, FunctionDefinition.operation field is still a String , so can anyone clarify whether we can use **operation** type as an Object  and pass Open API details inline ? or using it as a String and reference the swagger is the only option right now ?  

**Main version :**
<img width="1416" alt="Screenshot 2023-12-04 at 7 15 47 PM" src="https://github.com/serverlessworkflow/specification/assets/4058961/b7f9cca1-64ba-4798-ad54-84bd7b113368">

**0.8 version :**
<img width="1418" alt="Screenshot 2023-12-04 at 7 16 09 PM" src="https://github.com/serverlessworkflow/specification/assets/4058961/7c7015c0-9cbc-4a5e-9d74-f377e86eb199">

**Comment by cdavernas** [comment]
@shahbaaz31here Sorry, it seems we overlooked your question!

The `operation` field needs to be a `string`, it seems there was a bug with the schema.

I'm closing the discussion, though, as it applies to a discontinued version of the spec.
